File: Medium/2025-10-24-2048-Next-Greater-Numerically-Balanced-Number/solution.py
class Solution:
    def nextGreaterBalanced(self, n: int) -> int:
        def is_balanced(num):
            count = {}
            for digit in str(num):
                count[digit] = count.get(digit, 0) + 1
            for digit, cnt in count.items():
                if int(digit) != cnt:
                    return False
            return True

        n += 1
        while True:
            if is_balanced(n):
                return n
            n += 1

File: Medium/2025-10-24-2048-Next-Greater-Numerically-Balanced-Number/solution.js
function nextGreaterBalancedNumber(n) {
    const isBalanced = (num) => {
        const count = {};
        const strNum = num.toString();
        for (const char of strNum) {
            count[char] = (count[char] || 0) + 1;
        }
        for (const [digit, occurrences] of Object.entries(count)) {
            if (parseInt(digit) !== occurrences) {
                return false;
            }
        }
        return true;
    };

    let nextNum = n + 1;
    while (true) {
        if (isBalanced(nextNum)) {
            return nextNum;
        }
        nextNum++;
    }
}